sim/shaman/enhancement/rotation.go

```diff
@@ -75,22 +70,49 @@ func (rotation *AdaptiveRotation) DoAction(enh *EnhancementShaman, sim *core.Sim
 		return
 	}
 
+	if enh.Talents.MaelstromWeapon > 0 {
+		lbCastTime := enh.ApplyCastSpeed(time.Millisecond*2500 - time.Duration(500*enh.MaelstromWeaponAura.GetStacks()))
+		if enh.MaelstromWeaponAura.GetStacks() >= 3 {
+			if time.Duration(lbCastTime) < enh.AutoAttacks.NextAttackAt()+150*time.Millisecond {
+				if !enh.LightningBolt.Cast(sim, target) {
+					enh.WaitForMana(sim, enh.LightningBolt.CurCast.Cost)
+				}
+				return
+			}
+		}
+	}
```

Comment:
So, not a go genuis, but I'd have a hunch go has short curcuit evaluation.

So you can reduce your outer-if-statement. Also, I'd move `lbCastTime` calculation to when you know you have 3 maelstrom stacks.

```go
if enh.Talents.MaelstromWeapon > 0 && enh.MaelstromWeaponAura.GetStacks() >= 3 {
	lbCastTime := enh.ApplyCastSpeed(time.Millisecond*2500 - time.Duration(500*enh.MaelstromWeaponAura.GetStacks()))
	if time.Duration(lbCastTime) < enh.AutoAttacks.NextAttackAt()+150*time.Millisecond {
		if !enh.LightningBolt.Cast(sim, target) {
			enh.WaitForMana(sim, enh.LightningBolt.CurCast.Cost)
		}
		return
	}
}
```

Reply:
You prob should use `shaman.LightningBolt.DefaultCast.CastTime` instead of fixed `time.Millisecond * 2500` just in case somehow you had the cast speed reduction talent.

Reply:
im not sure if its possible to get that talent and maelstrom at the same time, but ill change it regardless, didnt know i could do it that way i just copied what the shadow priests did haha

Reply:
i double checked, you actually can! I think its only like 1-2 points, but you can!